src/backend/commands/copy.c

```diff
@@ -722,34 +722,57 @@ CopyGetData(CopyState cstate, void *databuf, int datasize)
 	switch (cstate->copy_dest)
 	{
 		case COPY_FILE:
-			bytesread = fread(databuf, 1, datasize, cstate->copy_file);
```

Comment:
do we need to replace fwrite as fread?

Reply:
`fwrite` only blocks if its internal buffer is full and it can't write more data. Here that happens only when the pipe is so slow it can't read even one row's worth of data before timeout runs out (since `fwrite` is called on every row). So `fwrite` blocking happens only under even more extreme conditions than `fread` blocking. Also, again, replacing `fwrite` with raw `write` would probably degrade performance (and I think here the effect would actually be noticeable)